core/client/src/notifications.rs

```diff
@@ -182,13 +295,23 @@ mod tests {
 	use runtime_primitives::testing::{H256 as Hash, Block as RawBlock, ExtrinsicWrapper};
 	use super::*;
 	use futures::Stream;
+	use ::std::iter::{empty, Empty};
 
+	type TestChangeSet = (Vec<(StorageKey, Option<StorageData>)>,
+		Vec<(StorageKey, Vec<(StorageKey, Option<StorageData>)>)>);
 	#[cfg(test)]
-	impl From<Vec<(StorageKey, Option<StorageData>)>> for StorageChangeSet {
-		fn from(changes: Vec<(StorageKey, Option<StorageData>)>) -> Self {
+	impl From<TestChangeSet> for StorageChangeSet {
+		fn from(changes: TestChangeSet) -> Self {
+			// warning hardcoded child trie wildcard to test upon
+			let child_filters = Some([
+				(StorageKey(vec![4]), None),
+				(StorageKey(vec![5]), None),
+			].into_iter().cloned().collect());
```

Comment:
why `into_iter().cloned().collect()` can't you just do `vec![...]` directly or is it a `HashSet`?

Reply:
yes HashMap